static/js/src/advantage/subscribe/react/components/BuyButton/BuyButton.tsx

```diff
@@ -145,157 +193,22 @@ function StepOne({ setStep, error, setError }: StepOneProps) {
-        ) {
-          setError(<>You already have trialled this product</>);
-        } else {
-          Sentry.captureException(error);
-          setError(
-            <>
-              Sorry, there was an unknown error with the free trial. Check the
-              details and try again. Contact{" "}
-              <a href="https://ubuntu.com/contact-us">Canonical sales</a> if the
-              problem persists.
-            </>
-          );
-        }
-      },
-    });
-  };
-
-  const trialNotAvailable = () => {
-    if (getIsFreeTrialEnabled()) {
-      return (
-        <Row>
-          <Col size={10} emptyLarge={2}>
-            <p>
-              <strong>
-                Free Trial is not available for this account.{" "}
-                <a href="/contact-us">Contact us</a> for further information.
-              </strong>
-            </p>
-          </Col>
-        </Row>
-      );
-    }
-    return null;
-  };
-
   return (
-    <>
-      <ModalHeader title="Your details" />
-
-      <ModalBody
-        isLoading={isUserInfoLoading || isProductLoading || isPreviewLoading}
-        error={error}
-      >
-        <>
-          <Summary />
-          {product?.canBeTrialled ? (
-            <FreeTrialRadio
-              isUsingFreeTrial={isUsingFreeTrial}
-              setIsUsingFreeTrial={setIsUsingFreeTrial}
-            />
-          ) : (
-            trialNotAvailable()
-          )}
-          <PaymentMethodSummary setStep={setStep} />
-          <Row className="u-no-padding">
-            <Col size={12}>
-              <CheckboxInput
-                name="TermsCheckbox"
-                onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-                  setTermsChecked(e.target.checked);
-                }}
-                label={
-                  <>
-                    I agree to the{" "}
-                    <a
-                      href="/legal/ubuntu-advantage-service-terms"
-                      target="_blank"
-                      rel="noopener noreferrer"
-                    >
-                      Ubuntu Advantage service terms
-                    </a>
-                  </>
-                }
-              />
-            </Col>
-          </Row>
-        </>
-      </ModalBody>
-
-      <ModalFooter>
-        <ActionButton
-          className="col-small-2 col-medium-2 col-3 u-no-margin"
-          appearance="positive"
-          style={{ textAlign: "center" }}
-          disabled={!areTermsChecked}
-          onClick={isUsingFreeTrial ? onStartTrialClick : onPayClick}
-          loading={
-            purchaseMutation.isLoading ||
-            freeTrialMutation.isLoading ||
-            isPendingPurchaseLoading
-          }
-        >
-          Buy
-        </ActionButton>
-      </ModalFooter>
-    </>
+    <ActionButton
+      className="col-small-2 col-medium-2 col-3 u-no-margin"
+      appearance="positive"
+      style={{ textAlign: "center" }}
+      disabled={!areTermsChecked}
```

Comment:
Seems like you should include loading states here as well to prevent errors? Correct me if I'm wrong, but this doesn't seem to be handled by ActionButton itself.

Reply:
A few lines down I define the loading state as
```
      loading={
        purchaseMutation.isLoading ||
        freeTrialMutation.isLoading ||
        isPendingPurchaseLoading
      }
```
isn't that what you mean?

Reply:
It seems that making it "loading" doesn't automatically make it "disabled"?

Reply:
The button isn't greyed out but you can't click it https://canonical-web-and-design.github.io/react-components/?path=/docs/actionbutton--default-story